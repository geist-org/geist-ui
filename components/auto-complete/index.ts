import AutoComplete from './auto-complete'
import AutoCompleteItem from './auto-complete-item'
import AutoCompleteSearching from './auto-complete-searching'
import AutoCompleteEmpty from './auto-complete-empty'

export type AutoCompleteComponentType = typeof AutoComplete & {
  Item: typeof AutoCompleteItem
  Option: typeof AutoCompleteItem
  Searching: typeof AutoCompleteSearching
  Empty: typeof AutoCompleteEmpty
}
;(AutoComplete as AutoCompleteComponentType).Item = AutoCompleteItem
;(AutoComplete as AutoCompleteComponentType).Option = AutoCompleteItem
;(AutoComplete as AutoCompleteComponentType).Searching = AutoCompleteSearching
;(AutoComplete as AutoCompleteComponentType).Empty = AutoCompleteEmpty

export type {
  AutoCompleteOption,
  AutoCompleteOptions,
  AutoCompleteProps,
  AutoCompleteTypes,
} from './auto-complete'
export default AutoComplete as AutoCompleteComponentType
