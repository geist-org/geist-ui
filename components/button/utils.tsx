import React, { ReactNode } from 'react'
import ButtonIcon from './button-icon'
import { ButtonProps } from './button'
import { ButtonGroupConfig } from '../button-group/button-group-context'
import useClasses from '../use-classes'

export const getButtonChildrenWithIcon = (
  auto: boolean,
  children: ReactNode,
  icons: {
    icon?: React.ReactNode
    iconRight?: React.ReactNode
  },
) => {
  const { icon, iconRight } = icons
  const hasIcon = icon || iconRight
  const isRight = Boolean(iconRight)
  const paddingForAutoMode = auto
    ? `calc(var(--geist-ui-button-height) / 2 + var(--geist-ui-button-icon-padding) * .5)`
    : 0
  const classes = useClasses('text', isRight ? 'right' : 'left')

  if (!hasIcon) return <div className="text">{children}</div>
  if (React.Children.count(children) === 0) {
    return (
      <ButtonIcon isRight={isRight} isSingle>
        {hasIcon}
      </ButtonIcon>
    )
  }
  return (
    <>
      <ButtonIcon isRight={isRight}>{hasIcon}</ButtonIcon>
      <div className={classes}>
        {children}
        <style jsx>{`
          .left {
            padding-left: ${paddingForAutoMode};
          }
          .right {
            padding-right: ${paddingForAutoMode};
          }
        `}</style>
      </div>
    </>
  )
}

export const filterPropsWithGroup = <T extends React.PropsWithChildren<ButtonProps>>(
  props: T,
  config: ButtonGroupConfig,
): T => {
  if (!config.isButtonGroup) return props
  return {
    ...props,
    auto: true,
    shadow: false,
    ghost: config.ghost || props.ghost,
    type: config.type || props.type,
    disabled: config.disabled || props.disabled,
  }
}
