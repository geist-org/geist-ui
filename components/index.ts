// @ts-ignore
/// <reference types="styled-jsx" />

export { default as AutoComplete } from './auto-complete'
export type { AutoCompleteProps } from './auto-complete'

export { default as Avatar } from './avatar'
export type { AvatarProps, AvatarGroupProps } from './avatar'

export { default as Badge } from './badge'
export type { BadgeProps, BadgeAnchorProps } from './badge'

export { default as Breadcrumbs } from './breadcrumbs'
export type {
  BreadcrumbsProps,
  BreadcrumbsItemProps,
  BreadcrumbsSeparatorProps,
} from './breadcrumbs'

export { default as Button } from './button'
export type { ButtonProps } from './button'

export { default as ButtonDropdown } from './button-dropdown'
export type { ButtonDropdownProps, ButtonDropdownItemProps } from './button-dropdown'

export { default as ButtonGroup } from './button-group'
export type { ButtonGroupProps } from './button-group'

export { default as Capacity } from './capacity'
export type { CapacityProps } from './capacity'

export { default as Card } from './card'
export type { CardProps, CardContentProps, CardFooterProps } from './card'

export { default as Checkbox } from './checkbox'
export type { CheckboxProps, CheckboxGroupProps } from './checkbox'

export { default as Code } from './code'
export type { CodeProps } from './code'

export { default as Collapse } from './collapse'
export type { CollapseProps, CollapseGroupProps } from './collapse'

export { default as Description } from './description'
export type { DescriptionProps } from './description'

export { default as Display } from './display'
export type { DisplayProps } from './display'

export { default as Divider } from './divider'
export type { DividerProps } from './divider'

export { default as Dot } from './dot'
export type { DotProps } from './dot'

export { default as Drawer } from './drawer'
export type { DrawerProps } from './drawer'

export { default as Fieldset } from './fieldset'
export type {
  FieldsetProps,
  FieldsetTitleProps,
  FieldsetSubtitleProps,
  FieldsetGroupProps,
  FieldsetFooterProps,
  FieldsetContentProps,
} from './fieldset'

export { default as GeistProvider } from './geist-provider'
export type { GeistProviderProps } from './geist-provider'

export { default as Grid } from './grid'
export type { GridProps, GridContainerProps } from './grid'

export { default as Image } from './image'
export type { ImageProps, ImageBrowserProps } from './image'

export { default as Input } from './input'
export type { InputProps, InputPasswordProps } from './input'

export { default as Keyboard } from './keyboard'
export type { KeyboardProps } from './keyboard'

export { default as Link } from './link'
export type { LinkProps } from './link'

export { default as Loading } from './loading'
export type { LoadingProps } from './loading'

export { default as Modal } from './modal'
export type {
  ModalProps,
  ModalTitleProps,
  ModalSubtitleProps,
  ModalContentProps,
  ModalActionProps,
} from './modal'

export { default as Note } from './note'
export type { NoteProps } from './note'

export { default as Page } from './page'
export type {
  PageProps,
  PageHeaderProps,
  PageContentProps,
  PageFooterProps,
} from './page'

export { default as Pagination } from './pagination'
export type {
  PaginationProps,
  PaginationNextProps,
  PaginationPreviousProps,
} from './pagination'

export { default as Popover } from './popover'
export type { PopoverProps, PopoverItemProps } from './popover'

export { default as Progress } from './progress'
export type { ProgressProps } from './progress'

export { default as Radio } from './radio'
export type { RadioProps, RadioGroupProps, RadioDescriptionProps } from './radio'

export { default as Rating } from './rating'
export type { RatingProps } from './rating'

export { default as Select } from './select'
export type { SelectProps, SelectOptionProps } from './select'

export { default as Slider } from './slider'
export type { SliderProps } from './slider'

export { default as Snippet } from './snippet'
export type { SnippetProps } from './snippet'

export { default as Spacer } from './spacer'
export type { SpacerProps } from './spacer'

export { default as Spinner } from './spinner'
export type { SpinnerProps } from './spinner'

export { default as Table } from './table'
export type { TableProps, TableColumnProps } from './table'

export { default as Tabs } from './tabs'
export type { TabsProps } from './tabs'

export { default as Tag } from './tag'
export type { TagProps } from './tag'

export { default as Text } from './text'
export type { TextProps } from './text'

export { default as Textarea } from './textarea'
export type { TextareaProps } from './textarea'

export { default as Themes } from './themes'
export type { GeistUIThemes, GeistUserTheme } from './themes'

export { default as Toggle } from './toggle'
export type { ToggleProps } from './toggle'

export { default as Tooltip } from './tooltip'
export type { TooltipProps } from './tooltip'

export { default as Tree } from './tree'
export type { TreeProps } from './tree'

export { default as useAllThemes } from './use-all-themes'
export type { AllThemesConfig } from './use-all-themes'

export { default as useToasts } from './use-toasts'
export type { Toast, ToastInput, ToastAction, ToastLayout } from './use-toasts'

export { default as User } from './user'
export type { UserProps } from './user'

export { default as useBodyScroll } from './use-body-scroll'
export type { BodyScrollOptions } from './use-body-scroll'

export { default as useClipboard } from './use-clipboard'
export type { UseClipboardOptions } from './use-clipboard'

export { default as useMediaQuery } from './use-media-query'
export type { ResponsiveOptions, ResponsiveBreakpoint } from './use-media-query'

export { default as useKeyboard, KeyMod, KeyCode } from './use-keyboard'
export type { KeyboardOptions, UseKeyboardHandler } from './use-keyboard'

export { default as useInput } from './use-input'
export { default as useModal } from './use-modal'
export { default as useTabs } from './use-tabs'
export { default as useClickAway } from './use-click-away'
export { default as useCurrentState } from './use-current-state'
export { default as CssBaseline } from './css-baseline'
export { default as useTheme } from './use-theme'
export { default as useClasses } from './use-classes'
export { default as useScale } from './use-scale'
export { withScale, ScalePropKeys, ScaleContext } from './use-scale'
export type {
  ScaleProps,
  ScaleConfig,
  GetScalePropsFunction,
  GetAllScalePropsFunction,
} from './use-scale'

export { useRect } from './utils/layouts'
export type { ReactiveDomReact } from './utils/layouts'
export { default as Highlight } from './shared/highlight'
export type { HighlightProps } from './shared/highlight'
