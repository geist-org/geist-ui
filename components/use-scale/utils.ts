import {
  GetAllScalePropsFunction,
  GetScalePropsFunction,
  ScaleProps,
  ScalePropKeys,
} from './scale-context'

export const generateGetScaleProps = <P>(
  props: P & ScaleProps,
): GetScalePropsFunction => {
  const getScaleProps: GetScalePropsFunction = keyOrKeys => {
    if (!Array.isArray(keyOrKeys)) return props[keyOrKeys as keyof ScaleProps]
    let value = undefined
    for (const key of keyOrKeys) {
      const currentValue = props[key]
      if (typeof currentValue !== 'undefined') {
        value = currentValue
      }
    }
    return value
  }
  return getScaleProps
}

export const generateGetAllScaleProps = <P>(
  props: P & ScaleProps,
): GetAllScalePropsFunction => {
  const getAllScaleProps: GetAllScalePropsFunction = () => {
    let scaleProps: ScaleProps = {}
    for (const key of ScalePropKeys) {
      const value = props[key as keyof ScaleProps]
      if (typeof value !== 'undefined') {
        scaleProps[key as keyof ScaleProps] = value as any
      }
    }
    return scaleProps
  }
  return getAllScaleProps
}
