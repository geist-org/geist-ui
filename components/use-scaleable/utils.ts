import { ScaleableProps } from './scaleable-context'

export type ScaleablePropsAndInvalid = keyof ScaleableProps | 'size'

export const ScaleablePropKeys: Array<ScaleablePropsAndInvalid> = [
  'paddingLeft',
  'pl',
  'paddingRight',
  'pr',
  'paddingTop',
  'pt',
  'paddingBottom',
  'pb',
  'marginTop',
  'mt',
  'marginRight',
  'mr',
  'marginBottom',
  'mb',
  'marginLeft',
  'ml',
  'px',
  'py',
  'mx',
  'my',
  'width',
  'height',
  'font',
  'unit',
  'scale',
  'size',
]

export const filterScaleableProps = <T extends Record<any, any>>(props: T) => {
  const keys = Object.keys(props).filter(key => key !== '')
  const nextProps: any = {}
  for (const key of keys) {
    if (!(ScaleablePropKeys as string[]).includes(key)) {
      nextProps[key] = props[key]
    }
  }
  return nextProps as Omit<T, ScaleablePropsAndInvalid>
}
