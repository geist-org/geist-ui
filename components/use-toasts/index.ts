import useToasts from './use-toast'

export type { ToastAction, Toast, ToastTypes, ToastInput, ToastLayout } from './use-toast'
export default useToasts
