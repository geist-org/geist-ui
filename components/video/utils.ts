export const formatTime = (seconds: number) => {
  const hh = Math.floor(seconds / 3600)
  const mm = Math.floor(seconds / 60) % 60
  const ss = Math.floor(seconds) % 60
  const pad = (num: number) => (num < 10 ? '0' + num : num)
  return `${hh ? pad(hh) + ':' : ''}${pad(mm)}:${pad(ss)}`
}

export const isFullscreenEnabled = () => {
  const elem = document as any
  return (
    elem.fullscreenEnabled ||
    elem.webkitFullscreenEnabled ||
    elem.mozFullScreenEnabled ||
    elem.msFullscreenEnabled
  )
}

export const openFullscreen = (element: HTMLElement) => {
  const elem = element as any
  if (elem.requestFullscreen) {
    elem.requestFullscreen()
  } else if (elem.mozRequestFullScreen) {
    elem.mozRequestFullScreen()
  } else if (elem.webkitRequestFullscreen) {
    elem.webkitRequestFullscreen()
  } else if (elem.msRequestFullscreen) {
    elem.msRequestFullscreen()
  }
}

export const closeFullscreen = () => {
  const elem = document as any
  if (elem.exitFullscreen) {
    elem.exitFullscreen()
  } else if (elem.mozCancelFullScreen) {
    elem.mozCancelFullScreen()
  } else if (elem.webkitExitFullscreen) {
    elem.webkitExitFullscreen()
  } else if (elem.msExitFullscreen) {
    elem.msExitFullscreen()
  }
}

export const onFullscreenChange = (func: (state: boolean) => void) => {
  const doc = document as any
  doc.addEventListener('fullscreenchange', () => func(!!doc.fullscreen), true)
  doc.addEventListener('mozfullscreenchange', () => func(!!doc.mozFullScreen), true)
  doc.addEventListener('webkitfullscreenchange', () => func(!!doc.webkitIsFullScreen), true)
  doc.addEventListener('msfullscreenchange', () => func(!!doc.msFullscreenElement), true)
}

export const isSafari = () =>
  navigator.userAgent.includes('Safari') && !navigator.userAgent.includes('Chrome')
