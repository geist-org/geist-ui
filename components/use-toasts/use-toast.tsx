import React, { CSSProperties, useEffect } from 'react'
import type { NormalTypes } from '../utils/prop-types'
import { defaultToastLayout, useGeistUIContext } from '../utils/use-geist-ui-context'
import { getId } from '../utils/collections'
import { ToastPlacement } from '../use-toasts/helpers'

export interface ToastAction {
  name: string
  handler: (event: React.MouseEvent<HTMLButtonElement>, cancel: () => void) => void
  passive?: boolean
}
export type ToastTypes = NormalTypes
export type ToastLayout = {
  padding?: CSSProperties['padding']
  margin?: CSSProperties['margin']
  width?: CSSProperties['width']
  maxWidth?: CSSProperties['maxWidth']
  maxHeight?: CSSProperties['maxHeight']
  placement?: ToastPlacement
}
export interface ToastInput {
  text: string | React.ReactNode
  type?: ToastTypes
  id?: string
  delay?: number
  actions?: Array<ToastAction>
}
export type ToastInstance = {
  visible: boolean
  cancel: () => void
  _timeout: null | number
  _internalIdent: string
}

export type Toast = Required<ToastInput> & ToastInstance

const defaultToast = {
  delay: 2000,
  type: 'default' as ToastTypes,
}

export type ToastHooksResult = {
  toasts: Array<Toast>
  setToast: (toast: ToastInput) => void
  removeAll: () => void
  findToastOneByID: (ident: string) => Toast | undefined
  removeToastOneByID: (ident: string) => void
}

const useToasts = (layout?: ToastLayout): ToastHooksResult => {
  const { updateToasts, toasts, updateToastLayout, updateLastToastId } =
    useGeistUIContext()

  useEffect(() => {
    if (!layout) return
    updateToastLayout(() =>
      layout
        ? {
            ...defaultToastLayout,
            ...layout,
          }
        : defaultToastLayout,
    )
  }, [])

  const cancel = (internalId: string) => {
    updateToasts((currentToasts: Array<Toast>) =>
      currentToasts.map(item => {
        if (item._internalIdent !== internalId) return item
        return { ...item, visible: false }
      }),
    )
    updateLastToastId(() => internalId)
  }
  const removeAll = () => {
    updateToasts(last => last.map(toast => ({ ...toast, visible: false })))
  }
  const findToastOneByID = (id: string) => toasts.find(t => t.id === id)
  const removeToastOneByID = (id: string) => {
    updateToasts(last =>
      last.map(toast => {
        if (toast.id !== id) return toast
        return {
          ...toast,
          visible: false,
        }
      }),
    )
  }

  const setToast = (toast: ToastInput): void => {
    const internalIdent = `toast-${getId()}`
    const delay = toast.delay || defaultToast.delay
    if (toast.id) {
      const hasIdent = toasts.find(t => t.id === toast.id)
      if (hasIdent) {
        throw new Error('Toast: Already have the same key: "ident"')
      }
    }

    updateToasts((last: Array<Toast>) => {
      const newToast: Toast = {
        delay,
        text: toast.text,
        visible: true,
        type: toast.type || defaultToast.type,
        id: toast.id || internalIdent,
        actions: toast.actions || [],
        _internalIdent: internalIdent,
        _timeout: window.setTimeout(() => {
          cancel(internalIdent)
          if (newToast._timeout) {
            window.clearTimeout(newToast._timeout)
            newToast._timeout = null
          }
        }, delay),
        cancel: () => cancel(internalIdent),
      }
      return [...last, newToast]
    })
  }

  return {
    toasts,
    setToast,
    removeAll,
    findToastOneByID,
    removeToastOneByID,
  }
}

export default useToasts
