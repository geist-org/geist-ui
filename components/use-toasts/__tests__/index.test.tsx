import React from 'react'
import { mount, ReactWrapper } from 'enzyme'
import { useToasts, GeistProvider } from 'components'
import { nativeEvent, updateWrapper } from 'tests/utils'
import { ToastInput } from '../use-toast'

const MockToast: React.FC<unknown> = () => {
  const { setToast } = useToasts()
  const clickHandler = (e: any = {}) => {
    const keys = ['text', 'delay', 'type', 'actions']
    const params = keys.reduce((pre, key) => {
      const value = e.target[key]
      if (!value) return pre
      return { ...pre, [key]: value }
    }, {})
    setToast(params as ToastInput)
  }
  return (
    <div id="btn" onClick={clickHandler}>
      btn
    </div>
  )
}

const triggerToast = (wrapper: ReactWrapper, params = {}) => {
  wrapper.find('#btn').simulate('click', {
    ...nativeEvent,
    target: params,
  })
}

const expectToastIsShow = (wrapper: ReactWrapper) => {
  const toast = wrapper.find('.toasts').find('.toast')
  expect(toast.length).not.toBe(0)
}

const expectToastIsHidden = (wrapper: ReactWrapper) => {
  const toast = wrapper.find('.toasts').find('.toast')
  expect(toast.length).toBe(0)
}

describe('UseToast', () => {
  it('should render correctly', async () => {
    const wrapper = mount(
      <GeistProvider>
        <MockToast />
      </GeistProvider>,
    )

    expectToastIsHidden(wrapper)
    triggerToast(wrapper, { text: 'test-value' })
    await updateWrapper(wrapper)
    expectToastIsShow(wrapper)
  })

  it('should work with different types', async () => {
    const wrapper = mount(
      <GeistProvider>
        <MockToast />
      </GeistProvider>,
    )

    expectToastIsHidden(wrapper)
    triggerToast(wrapper, { type: 'success', text: 'hello' })
    await updateWrapper(wrapper, 100)
    expectToastIsShow(wrapper)
    expect(wrapper.find('.toasts').html()).toMatchSnapshot()
  })

  it('should close toast', async () => {
    const wrapper = mount(
      <GeistProvider>
        <MockToast />
      </GeistProvider>,
    )

    expectToastIsHidden(wrapper)
    triggerToast(wrapper, { delay: 100, text: 'hello' })
    await updateWrapper(wrapper, 0)
    expectToastIsShow(wrapper)
    // Element already hidden, but Dom was removed after delay
    await updateWrapper(wrapper, 350)
    const toast = wrapper.find('.toasts').find('.toast')
    expect(toast.length).not.toBe(0)
  })

  it('the removeal should be delayed when hover is triggerd', async () => {
    const wrapper = mount(
      <GeistProvider>
        <MockToast />
      </GeistProvider>,
    )

    expectToastIsHidden(wrapper)
    triggerToast(wrapper, { delay: 100, text: 'hello' })
    await updateWrapper(wrapper, 0)
    expectToastIsShow(wrapper)

    wrapper.find('.toasts').simulate('mouseEnter', nativeEvent)
    await updateWrapper(wrapper, 350)

    // Hover event will postpone hidden event
    let toast = wrapper.find('.toasts').find('.toast')
    expect(toast.length).not.toBe(0)

    // Restart hidden event after mouse leave
    wrapper.find('.toasts').simulate('mouseLeave', nativeEvent)
    await updateWrapper(wrapper, 350 + 200)
    toast = wrapper.find('.toasts').find('.toast')
    expect(toast.length).toBe(0)
  })

  it('should render different actions', async () => {
    const wrapper = mount(
      <GeistProvider>
        <MockToast />
      </GeistProvider>,
    )
    const actions = [
      {
        name: 'remove',
        handler: () => {},
      },
      {
        name: 'remove',
        handler: () => {},
        passive: true,
      },
    ]

    triggerToast(wrapper, { actions, text: 'hello' })
    await updateWrapper(wrapper)
    expectToastIsShow(wrapper)
    expect(wrapper.find('.toasts').html()).toMatchSnapshot()
  })

  it('should close toast when action triggered', async () => {
    const wrapper = mount(
      <GeistProvider>
        <MockToast />
      </GeistProvider>,
    )
    const actions = [
      {
        name: 'remove',
        handler: (_event: any, cancel: () => void) => cancel(),
      },
    ]

    expectToastIsHidden(wrapper)
    triggerToast(wrapper, { actions, text: 'hello' })
    await updateWrapper(wrapper)
    expectToastIsShow(wrapper)
    wrapper.find('.action').find('.btn').at(0).simulate('click', nativeEvent)

    // Element already hidden, but Dom was removed after delay
    await updateWrapper(wrapper, 250)
    const toast = wrapper.find('.toasts').find('.toast')
    expect(toast.length).not.toBe(0)
  })

  it('should work with multiple toasts', async () => {
    const wrapper = mount(
      <GeistProvider>
        <MockToast />
      </GeistProvider>,
    )

    expectToastIsHidden(wrapper)
    triggerToast(wrapper, { delay: 100, text: 'hello' })
    triggerToast(wrapper, { delay: 100, text: 'hello' })
    triggerToast(wrapper, { delay: 100, text: 'hello' })
    triggerToast(wrapper, { delay: 100, text: 'hello' })
    triggerToast(wrapper, { delay: 100, text: 'hello' })
    triggerToast(wrapper, { delay: 200, text: 'hello' })

    /**
     * If there are multiple Toasts at different deplay in the stack,
     * the destory Dom event will wait for the maximum delay time.
     */
    await updateWrapper(wrapper, 350)
    expectToastIsShow(wrapper)

    await updateWrapper(wrapper, 200)
    const toast = wrapper.find('.toasts').find('.toast')
    expect(toast.length).not.toBe(0)
  })
})
