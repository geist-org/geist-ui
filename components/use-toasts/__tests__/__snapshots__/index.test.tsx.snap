// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UseToast should render different actions 1`] = `
"<div class=\\"toasts\\"><div class=\\"toast  toast-enter\\"><div class=\\"message\\">hello</div><div class=\\"action\\"><button type=\\"button\\" class=\\"btn\\"><div class=\\"text\\">remove</div><style>
          .btn {
            box-sizing: border-box;
            display: inline-block;
            line-height: calc(1.6666666666666665 * 16px);
            border-radius: 6px;
            font-weight: 400;
            font-size: 13px;
            user-select: none;
            outline: none;
            text-transform: capitalize;
            justify-content: center;
            text-align: center;
            white-space: nowrap;
            transition: background-color 200ms ease 0ms, box-shadow 200ms ease 0ms,
              border 200ms ease 0ms, color 200ms ease 0ms;
            position: relative;
            overflow: hidden;
            color: #fff;
            background-color: #000;
            border: 1px solid #000;
            cursor: pointer;
            pointer-events: auto;
            box-shadow: none;
            --geist-ui-button-icon-padding: calc(0.48466666666666663 * 16px);
            --geist-ui-button-height: calc(1.6666666666666665 * 16px);
            --geist-ui-button-color: #fff;
            --geist-ui-button-bg: #000;
            min-width: min-content;
            width: auto;
            height: calc(1.6666666666666665 * 16px);
            padding: 0 calc(0.7666666666666666 * 16px) 0 calc(0.7666666666666666 * 16px);
            margin: 0 0 0 0;
          }

          .btn:hover,
          .btn:focus {
            color: #000;
            --geist-ui-button-color: #000;
            background-color: #fff;
            border-color: #000;
            cursor: pointer;
            pointer-events: auto;
            box-shadow: none;
            transform: translate3d(0px, 0px, 0px);
          }

          .btn :global(.text) {
            position: relative;
            z-index: 1;
            display: inline-flex;
            justify-content: center;
            align-items: center;
            text-align: center;
            line-height: inherit;
            top: -1px;
          }

          .btn :global(.text p),
          .btn :global(.text pre),
          .btn :global(.text div) {
            margin: 0;
          }
        </style></button><button type=\\"button\\" class=\\"btn\\"><div class=\\"text\\">remove</div><style>
          .btn {
            box-sizing: border-box;
            display: inline-block;
            line-height: calc(1.6666666666666665 * 16px);
            border-radius: 6px;
            font-weight: 400;
            font-size: 13px;
            user-select: none;
            outline: none;
            text-transform: capitalize;
            justify-content: center;
            text-align: center;
            white-space: nowrap;
            transition: background-color 200ms ease 0ms, box-shadow 200ms ease 0ms,
              border 200ms ease 0ms, color 200ms ease 0ms;
            position: relative;
            overflow: hidden;
            color: #666;
            background-color: #fff;
            border: 1px solid #eaeaea;
            cursor: pointer;
            pointer-events: auto;
            box-shadow: none;
            --geist-ui-button-icon-padding: calc(0.48466666666666663 * 16px);
            --geist-ui-button-height: calc(1.6666666666666665 * 16px);
            --geist-ui-button-color: #666;
            --geist-ui-button-bg: #fff;
            min-width: min-content;
            width: auto;
            height: calc(1.6666666666666665 * 16px);
            padding: 0 calc(0.7666666666666666 * 16px) 0 calc(0.7666666666666666 * 16px);
            margin: 0 0 0 0;
          }

          .btn:hover,
          .btn:focus {
            color: #000;
            --geist-ui-button-color: #000;
            background-color: #fff;
            border-color: #000;
            cursor: pointer;
            pointer-events: auto;
            box-shadow: none;
            transform: translate3d(0px, 0px, 0px);
          }

          .btn :global(.text) {
            position: relative;
            z-index: 1;
            display: inline-flex;
            justify-content: center;
            align-items: center;
            text-align: center;
            line-height: inherit;
            top: -1px;
          }

          .btn :global(.text p),
          .btn :global(.text pre),
          .btn :global(.text div) {
            margin: 0;
          }
        </style></button></div><style>
          .toast {
            width: 420px;
            max-width: 90vw;
            max-height: 75px;
            display: flex;
            justify-content: space-between;
            align-items: center;
            color: #000;
            background-color: #fff;
            color: #000;
            border: 0;
            border-radius: 6px;
            opacity: 1;
            box-shadow: 0 5px 10px rgba(0, 0, 0, 0.12);
            transition: all 350ms cubic-bezier(0.1, 0.2, 0.1, 1);
            overflow: hidden;
          }
          .message {
            align-items: center;
            height: 100%;
            font-size: 0.875em;
            display: -webkit-box;
            word-break: break-all;
            padding-right: 8pt;
            overflow: hidden;
            max-height: 100%;
            text-overflow: ellipsis;
            -webkit-box-orient: vertical;
            -webkit-line-clamp: 2;
            line-height: 1.1rem;
          }
          .toast-enter {
            opacity: 0;
            height: 0;
            padding: 0;
            margin: 0;
            transform: translate(60px, 60px);
          }
          .toast-enter-active {
            opacity: 1;
            height: auto;
            margin: 8px 0;
            padding: 12px 16px;
            transform: translate(0, 0);
          }
          .toast-leave {
            opacity: 1;
            transform: translate(0, 0);
            height: auto;
            margin: 8px 0;
            padding: 12px 16px;
          }
          .toast-leave-active {
            opacity: 0;
            transform: translate(50px, -15px) scale(0.85);
          }
        </style></div><style>
        .toasts {
          position: fixed;
          width: auto;
          max-width: 100%;
          right: 16pt;
          bottom: 16pt;
          z-index: 2000;
          transition: all 400ms ease;
          box-sizing: border-box;
          display: flex;
          flex-direction: column;
        }
        .top {
          bottom: unset;
          flex-direction: column-reverse;
          top: 16pt;
        }
        .left {
          right: unset;
          left: 16pt;
        }
      </style></div>"
`;

exports[`UseToast should work with different types 1`] = `
"<div class=\\"toasts\\"><div class=\\"toast  toast-enter toast-enter-active\\"><div class=\\"message\\">hello</div><div class=\\"action\\"></div><style>
          .toast {
            width: 420px;
            max-width: 90vw;
            max-height: 75px;
            display: flex;
            justify-content: space-between;
            align-items: center;
            color: #000;
            background-color: #0070f3;
            color: white;
            border: 0;
            border-radius: 6px;
            opacity: 1;
            box-shadow: 0 5px 10px rgba(0, 0, 0, 0.12);
            transition: all 350ms cubic-bezier(0.1, 0.2, 0.1, 1);
            overflow: hidden;
          }
          .message {
            align-items: center;
            height: 100%;
            font-size: 0.875em;
            display: -webkit-box;
            word-break: break-all;
            padding-right: 8pt;
            overflow: hidden;
            max-height: 100%;
            text-overflow: ellipsis;
            -webkit-box-orient: vertical;
            -webkit-line-clamp: 2;
            line-height: 1.1rem;
          }
          .toast-enter {
            opacity: 0;
            height: 0;
            padding: 0;
            margin: 0;
            transform: translate(60px, 60px);
          }
          .toast-enter-active {
            opacity: 1;
            height: auto;
            margin: 8px 0;
            padding: 12px 16px;
            transform: translate(0, 0);
          }
          .toast-leave {
            opacity: 1;
            transform: translate(0, 0);
            height: auto;
            margin: 8px 0;
            padding: 12px 16px;
          }
          .toast-leave-active {
            opacity: 0;
            transform: translate(50px, -15px) scale(0.85);
          }
        </style></div><style>
        .toasts {
          position: fixed;
          width: auto;
          max-width: 100%;
          right: 16pt;
          bottom: 16pt;
          z-index: 2000;
          transition: all 400ms ease;
          box-sizing: border-box;
          display: flex;
          flex-direction: column;
        }
        .top {
          bottom: unset;
          flex-direction: column-reverse;
          top: 16pt;
        }
        .left {
          right: unset;
          left: 16pt;
        }
      </style></div>"
`;
